src/tests/badge.rs

```diff
@@ -25,14 +25,35 @@ struct BadgeRef {
     maintenance_attributes: HashMap<String, String>,
 }
 
-fn set_up() -> (DieselConnection, Crate, BadgeRef) {
-    let (_b, app, _middle) = app();
+struct BadgeTestCrate {
+    app: TestApp,
+    krate: Crate,
+}
 
-    let krate = {
-        let conn = app.diesel_database.get().unwrap();
-        let u = new_user("foo").create_or_update(&conn).unwrap();
-        CrateBuilder::new("badged_crate", u.id).expect_build(&conn)
-    };
+impl BadgeTestCrate {
+    /// Update the crate with badges, returning invalid badges
+    fn update(&self, badges: &HashMap<String, HashMap<String, String>>) -> Vec<String> {
+        self.app
+            .db(|conn| Badge::update_crate(conn, &self.krate, Some(badges)).unwrap())
+    }
+
+    /// Update the crate with badges, returning invalid badges
```

Comment:
Shouldn't this comment say something like "Update the crate to have no badges", rather than the same comment as `update`?